src/main/java/com/example/imagescaler/upscale/apiprovider/UpscaleApiProviderImpl.java

```diff
@@ -0,0 +1,52 @@
+package com.example.imagescaler.upscale.apiprovider;
+
+import com.example.imagescaler.upscale.configuration.ApiProperties;
+import java.time.Duration;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.http.client.reactive.ReactorClientHttpConnector;
+import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.BodyInserters;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.core.publisher.Mono;
+import reactor.netty.http.client.HttpClient;
+
+@Service
+public class UpscaleApiProviderImpl implements UpscaleApiProvider {
+  private final ApiProperties apiProperties;
+  private final WebClient webClient;
+
+  public UpscaleApiProviderImpl(
+    ApiProperties apiProperties,
+    WebClient.Builder webClientBuilder
+  ) {
+    this.apiProperties = apiProperties;
+    this.webClient = initializeWebClient(webClientBuilder);
+  }
+
+  private WebClient initializeWebClient(WebClient.Builder webClientBuilder) {
+    HttpClient httpClient = HttpClient
```

Comment:
Always prefer `final` vars, so that you're sure you're not going to reassign them later.

Lombok provides you with a handy keyword: `val`, so this should be `val httpClient`.